django/core/cache/backends/filebased.py

```diff
@@ -61,6 +61,14 @@ def set(self, key, value, timeout=DEFAULT_TIMEOUT, version=None):
             if not renamed:
                 os.remove(tmp_path)
 
+    def touch(self, key, timeout=DEFAULT_TIMEOUT, version=None):
+        if not self.has_key(key, version):
+            return False
+        else:
+            value = self.get(key, version=version)
+            self.set(key, value, timeout, version)
```

Comment:
isn't this still subject to the race condition discussed in implementation on the base backend? something else could `set` the key in between the `get` and `set`.